addons/account/models/account_analytic_line.py

```diff
@@ -80,50 +83,78 @@ def action_view_vendor_bill(self):
+        best_applicability = {'applicability': None, 'score': 0}
+        for rule in self.applicability_ids:
+            rule_score = 0
+            if domain and rule.domain == domain:
+                rule_score += 1
+            if domain and rule.domain != domain:
+                continue
+            if rule.account_prefix and account and account.code.startswith(rule.account_prefix):
+                rule_score += 1
+            elif rule.account_prefix and account and not account.code.startswith(rule.account_prefix):
+                continue
+            if rule.product_categ_id and product and rule.product_categ_id == product.categ_id:
+                rule_score += 1
+            elif rule.product_categ_id and product and rule.product_categ_id != product.categ_id:
+                continue
+            if rule_score > best_applicability['score']:
+                best_applicability = {'applicability': rule.applicability, 'score': rule_score}
+        return best_applicability['applicability'] or self.default_applicability
+
+
+class AccountAnalyticApplicability(models.Model):
+    _inherit = 'account.analytic.applicability'
+    _description = "Analytic Plan's Applicabilities"
+    account_prefix = fields.Char(string='Account Prefix', default='',
+                                 help="Prefix that define which accounts this applicability should apply on.")
+    product_categ_id = fields.Many2one('product.category', string='Product Category')
 
 
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
     _description = 'Analytic Line'
 
     product_id = fields.Many2one('product.product', string='Product', check_company=True)
-    general_account_id = fields.Many2one('account.account', string='Financial Account', ondelete='restrict', readonly=True,
-                                         related='move_id.account_id', store=True, domain="[('deprecated', '=', False), ('company_id', '=', company_id)]",
-                                         compute_sudo=True)
+    general_account_id = fields.Many2one('account.account', string='Financial Account', ondelete='restrict',
+                                         domain="[('deprecated', '=', False), ('company_id', '=', company_id)]",
+                                         readonly=False, compute="_compute_on_move_id", store=True)
+    journal_id = fields.Many2one('account.journal', string='Financial Journal', check_company=True,
+                                 readonly=False, compute="_compute_on_move_id", store=True)
     move_id = fields.Many2one('account.move.line', string='Journal Item', ondelete='cascade', index=True, check_company=True)
     code = fields.Char(size=8)
     ref = fields.Char(string='Ref.')
     category = fields.Selection(selection_add=[('invoice', 'Customer Invoice'), ('vendor_bill', 'Vendor Bill')])
 
+    @api.depends('move_id')
+    def _compute_on_move_id(self):
+        #TODO Check partner_id if stable with other modules
```

Comment:
what do you mean?

It shouldn't be `_compute_on_move_id`, but `_compute_partner_id`, `_compute_journal_id`,.. and then you can add dependencies in the next modules, changing the computation if needed